File: Backend/src/main/java/org/example/backend/domain/ScenePart.java
package org.example.backend.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "scene_part")
public class ScenePart {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ColumnDefault("nextval('scene_part_id_seq')")
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "name", nullable = false, length = 200)
    private String name;

    @Column(name = "news", nullable = false)
    private Boolean news = false;

    @Setter
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @ColumnDefault("nextval('scene_part_scene_id_seq')")
    @JoinColumn(name = "scene_id", nullable = false)
    private Scene scene;

}

File: TTSolver/Model/ScenePart.py
class ScenePart:
    def __init__(self, id: int, name: str, news: bool, scene: 'Scene'):
        self.id = id
        self.name = name
        self.news = news
        self.scene = scene

    def __str__(self):
        return f"ScenePart(id={self.id}, name={self.name}, news={self.news}, scene={self.scene})"
